executor/src/subscriber.rs

```diff
@@ -185,29 +84,26 @@ impl Subscriber {
             tokio::select! {
                 // Receive the ordered sequence of consensus messages from a consensus node.
                 Some(message) = self.rx_consensus.recv() => {
-                    // Process the consensus message (synchronize missing messages, download transaction data).
-                    let sequence = self.handle_consensus_message(&message).await?;
-
-                    // Update the latest consensus index. The state will atomically persist the change when
-                    // executing the transaction. It is important to increment the consensus index before
-                    // deserializing the transaction data because the consensus core will increment its own
-                    // index regardless of deserialization or other application-specific failures.
-                    self.next_consensus_index += sequence.len() as SequenceNumber;
-
-                    // Wait for the transaction data to be available in the store. We will then execute the transactions.
-                    for message in sequence {
-                        let digests = message.certificate.header.payload.keys().cloned().collect();
-                        let future = Self::waiter(digests, self.store.clone(), message);
-                        waiting.push(future);
-                    }
+                    // Send the certificate to the batch loader to download all transactions' data.
+                    self.tx_batch_loader
+                        .send(message.clone())
+                        .await
+                        .expect("Failed to send message ot batch loader");
+
+                    // Wait for the transaction data to be available in the store. We will then forward these
+                    // transactions to the Executor Core for execution.
```

Comment:
Maybe tell us a little more in this comment how this is guaranteed to happen, and therefore liveness if safe?